nibabies/workflows/bold/alignment.py

```diff
@@ -0,0 +1,310 @@
+        (roi2atlas, bold_mask_roi, [("out_file", "mask_file")]),
+        (roi2atlas, mul_roi, [("out_file", "in_file")]),
+        (parse_labels, mul_roi, [("label_ids", "operand_value")]),
+        (split_atlas_rois, mul_atlas_roi, [("out_files", "in_file")]),
+        (parse_labels, mul_atlas_roi, [("label_ids", "operand_value")]),
+        (mul_roi, vol_label, [("out_file", "in_file")]),
+        (atlas_labels, vol_label, [("out_file", "label_list_file")]),
+        (mul_atlas_roi, vol_atlas_label, [("out_file", "in_file")]),
+        (atlas_labels, vol_atlas_label, [("out_file", "label_list_file")]),
+        (bold_mask_roi, create_dtseries, [("out_file", "volume_data")]),
+        (vol_label, create_dtseries, [("out_file", "volume_structure_labels")]),
+        (vol_atlas_label, create_label, [
+            ("out_file", "volume_label"),
+            ("out_file", "structure_label_volume")]),
+        (create_dtseries, dilate, [("out_file", "in_file")]),
+        (dilate, resample, [("out_file", "in_file")]),
+        (create_label, resample, [("out_file", "template")]),
+        (resample, smooth, [("out_file", "in_file")]),
+        (smooth, separate, [("out_file", "in_file")]),
+        # end loop
+        (mul_roi, fmt_agg_rois, [("out_file", "rois")]),
+        (fmt_agg_rois, agg_rois, [
+            ("first_image", "in_file"),
+            ("op_files", "operand_files"),
+            ("op_string", "op_string")]),
+        (separate, combine_rois, [("volume_all_file", "in_files")]),
+        (combine_rois, outputnode, [("out", "subcortical_file")]),
+    ])
+    # fmt: on
+    return workflow
+
+
+def parse_roi_labels(label_file):
+    """
+    Parse a label file composed of one or more sets of:
+    <labelname>
+    <key> <red> <green> <blue> <alpha>
+
+    Return a list of structure names and label keys.
+
+    >>> structs, ids = parse_roi_labels(test_data / "labelfile.txt")
+    >>> structs
+    ['CEREBELLUM_LEFT', 'THALAMUS_LEFT', 'CAUDATE_LEFT']
+    >>> ids
+    [8, 10, 11]
+    """
+
+    with open(label_file) as fp:
+        lines = fp.readlines()
+    if len(lines) % 2 == 1:
+        raise RuntimeError("Label file is incomplete or invalid")
+    structs, label_ids = [], []
+    for idx, line in enumerate(lines):
+        if idx % 2 == 0:
+            structs.append(line.strip())
+        else:
+            label_ids.append(int(line.split(' ', 1)[0]))
+    return structs, label_ids
+
+
+def format_volume_rois(structs, rois):
+    """Format volume arguments for CiftiCreateDenseFromTemplate."""
+    return [(struct, roi) for struct, roi in zip(structs, rois)]
+
+
+def format_agg_rois(rois):
+    """
+    Helper function to format MultiImageMaths command.
+
+    Parameters
+    ----------
+    rois : `list` of `str`s
+        List of files
+
+    Returns
+    -------
+    first_image
+    op_files
+    op_string
+
+    """
+    return rois[0], rois[1:], ("-add %s " * (len(rois) - 1)).strip()
+
+
+def merge_rois(in_files):
```

Comment:
This feels like a significant enough function to write a test for. 3 `(3,1,1,5)` time series would exercise it plenty, and easy to modify to hit each error condition.